bConnect/functions/Set-bConnectEndpoint.ps1

```
@@ -0,0 +1,128 @@
+
+     .PARAMETER Owner
+         The new Owner of the Device. Is currently only Working for Mobile and MAC Endpoints
+
+     .PARAMETER ComplianceCheckCategory
+         The new ComplianceCheckCategory of the Device. Is currently only Working for Mobile and MAC Endpoints
+
+     .OUTPUTS
+         Endpoint
+
+     .NOTES
+         Additional information about the function.
+ #>
+Function Set-bConnectEndpoint
+{
+    [CmdletBinding(DefaultParameterSetName = 'WindowsEndpoint',
+        ConfirmImpact = 'Medium',
+        SupportsShouldProcess = $true)]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification = "Test-PSFShouldProcess is used instead of ShouldProcess.")]
+    Param
+    (
+        [Parameter(Mandatory = $true,
+            ValueFromPipelineByPropertyName = $true)]
+        [PsfValidatePattern('\b[A-F0-9]{8}(?:-[A-F0-9]{4}){3}-[A-F0-9]{12}\b', ErrorMessage = 'Failed to parse input as guid: {0}')]
+        $EndpointGuid,
+        $DisplayName,
+        $GuidOrgUnit,
+        [Parameter(ParameterSetName = 'MacEndpoint')]
+        [Parameter(ParameterSetName = 'WindowsEndpoint')]
+        $HostName,
+        [Parameter(ParameterSetName = 'WindowsEndpoint')]
+        $PrimaryMAC,
+        [Parameter(ParameterSetName = 'WindowsEndpoint')]
+        $Domain,
+        [Parameter(ParameterSetName = 'WindowsEndpoint')]
+        $GuidBootEnvironment,
+        [Parameter(ParameterSetName = 'WindowsEndpoint')]
+        $PublicKey,
+        [Parameter(ParameterSetName = 'WindowsEndpoint')]
+        $Comments,
+        $PrimaryUser,
+        [Parameter(ParameterSetName = 'MobileEndpoint')]
+        $Owner,
+        [Parameter(ParameterSetName = 'MobileEndpoint')]
+        $ComplianceCheckCategory
+    )
+
+    Begin
+    {
+        Assert-bConnectConnection
+    }
+    Process
+    {
+        ForEach ($endpointItem In $EndpointGuid)
+        {
+            If (Test-Guid $EndpointGuid)
+            {
```

This test is redundant, as the
```
[PsfValidatePattern('\b[A-F0-9]{8}(?:-[A-F0-9]{4}){3}-[A-F0-9]{12}\b', ErrorMessage = 'Failed to parse input as guid: {0}')]
```
Already handles validation.

As a general advice on handling this kind of test though:
- Consider what you want to do if it fails - there is no `else` block. It would silently ignore the input if it weren't a valid guid.
- Rather than testing for success, you should prefer testing for failure instead. As it is, an `else` would be below the large code that happens if conditions meet. This makes it a lot harder to trace back what decision point led the execution to that line (leading to lots of scrolling and headache). If instead you'd test whether the condition is _not_ true, you could in the `if` script handle termination (Using `Stop-PSFFunction -Continue` in this instance and maybe `continue` or `throw` in case of code that doesn't use my framework). That way you have condition and reaction in one neat block and save yourself one indentation level for the production code below (since the condition handles termination, we don't need to put it in an `else` block).

---

I removed the Test-GUID because it is not needed. I used the Edit-bConnectEndpoint as a template for this function and forgot to remove it. Thanks!